### ISSUE TITLE
False positive of `reportIncompatibleMethodOverride` for methods marked with `@staticmethod`

### ISSUE THREAD

**Issue by mickvangelderen** [microsoft/pyright#9718]
It seems to me that for static methods of the same name in an inheritance tree, the type signature used that function should not matter. However, `pyright` reports `reportIncompatibleMethodOverride`. 

Example with incompatible parameter types:

```py
class Base:
    @staticmethod
    def f(x: str): ...


class Derived(Base):
    @staticmethod
    def f(x: int): ...
#       ^ reportIncompatibleMethodOverride
```

Example with incompatible return types:

```py
class Base:
    @staticmethod
    def f() -> str:
        return ""


class Derived(Base):
    @staticmethod
    def f() -> int: 
#       ^ reportIncompatibleMethodOverride
        return 0
```

Static methods should only be called as `Base.f(...)` or `Derived.f(...)`. The class just serves as a sort of namespace. I understand that in Python there are no true static methods, but doesn't `@staticmethod` mean that methods decorated by it should be treated as such? Perhaps I need to adjust my mental model. 

Using `pyright` version `v1.1.392.post0`.

**Comment by erictraut** [comment]
Static methods do matter because `type[T]` is covariant. That means `Derived` is a subtype of `type[Base]`, and when you call `f()` on a value of type `type[Base]`, the return type is expected to be `str`. There has been some recent discussion in the typing forums to change the subtyping rules for `type`, but given the current way it's defined in the type specification, pyright's behavior is correct here.

**Comment by mickvangelderen** [comment]
So the following is a legitimate use-case of static methods?

```py
class Base:
    @staticmethod
    def f() -> str:
        return "base"


class Derived(Base):
    @staticmethod
    def f() -> int:
        return 0


def call_f(x: Base) -> str:
    return type(x).f()


# This assertion fails because `call_f` ends up calling `Derived.f` which does not return a `str`.
assert isinstance(call_f(Derived()), str)
```
